client/src/components/generic/DatePickerFilter.tsx

```diff
@@ -0,0 +1,255 @@
+  },
+
+  'Last 7 Days': {
+    startDate: function startDate(now: moment.Moment) {
+      return now.add(-7, 'days');
+    },
+    endDate: function endDate(now: moment.Moment) {
+      return now.add(-1, 'days');
+    }
+  },
+
+  'Last 30 Days': {
+    startDate: function startDate(now: moment.Moment) {
+      return now.add(-30, 'days');
+    },
+    endDate: function endDate(now: moment.Moment) {
+      return now.add(-1, 'days');
+    }
+  }
+};
+
+const styles = {
+    button: {
+        userSelect: 'none',
+    } as React.CSSProperties,
+    container: {
+        position: 'relative',
+        float: 'left',
+        zIndex: 17,
+    } as React.CSSProperties,
+    animateOpen: {
+        transition: '.3s',
+        transform: 'scale(1.0,1.0)',
+        transitionTimingFunction: 'cubic-bezier(0.4, 0, 0.2, 1)',
+    } as React.CSSProperties,
+    animateClose: {
+        transform: 'scale(1.0,0)',
+        transition: '0s',
+    } as React.CSSProperties,
+    list: {
+        position: 'absolute',
+        top: '0px',
+        left: '0px',
+    } as React.CSSProperties
+};
+
+// using styles from the select field menu
+const classNames = {
+    menu: ['md-inline-block', 'md-menu-container', 'md-menu-container--menu-below', 'md-select-field-menu',
+        'md-select-field-menu--stretch', 'md-select-field--toolbar', ''],
+    label: ['md-floating-label', 'md-floating-label--floating', ''],
+};
+
+export default class DatePickerFilter extends GenericComponent<any, any> {
+
+    static defaultProps = {
+        title: 'Timespan',
+        subtitle: 'Select range',
+        icon: 'more_vert'
+    };
+
+    constructor(props: any) {
+        super(props);
+
+        this.handleSelect = this.handleSelect.bind(this);
+        this.showOverlay = this.showOverlay.bind(this);
+        this.hideOverlay = this.hideOverlay.bind(this);
+
+        this.state = {
+            visible: false,
+            selectedValue: ''
+        };
+    }
+
+    /**
+     * Handle new dateRange selection
+     * @param {Object} newDateRange
+     */
+    handleSelect(newDateRange: any) {
+        let selectedValue = this.toISODateRange(newDateRange);
+
+        if (this.state.selectedValue !== selectedValue) {
+            this.setState({
+                selectedValue
+            });
+        }
+    }
+
+    showOverlay() {
+        this.setState({
+            visible: true
+        });
+    }
+
+    hideOverlay() {
+        this.setState({
+            visible: false
+        });
+        this.trigger('onChange', this.state.selectedValue);
+    }
+
+    /**
+     * Construct ISO8601 Date interval string from two dates
+     * startDate is before than endDate
+     *
+     * @param {moment} startDate
+     * @param {moment} endDate
+     * @returns {string} ISO8601 Date interval string
+     */
+    toISODateRange(dateRange: any) {
+        let {startDate, endDate} = dateRange;
+        let startDateISOString = `${startDate.format('YYYY-MM-DD')}T00:00:00.000Z`;
```

Comment:
You can use:

For local GMT:

```js
const start = moment().startOf('day'); // set to 12:00 am today
const end = moment().endOf('day'); // set to 23:59 pm today
```

For UTC:

```js
const start = moment.utc().startOf('day'); 
const end = moment.utc().endOf('day'); 
```

Reply:
the datepicker returns a non utc endOfDay and a non utc startOfDay.

If I apply utc startofday/endofday, depending on the timezone and the currenttime you can get the day before/after startofday/endofday..

So, to get the endofday/startofday for the day that the user picked i have to pick the localtime date and append the endofday/startofday time.

```javascript
    toISODateRange(dateRange: any) {
        let {startDate, endDate} = dateRange;

        let startDateISOString = `${startDate.format('YYYY-MM-DD')}T00:00:00.000Z`;
        console.log('#####################');
        console.log('startDate');
        console.log(startDate.toISOString());
        console.log('startDateISOString');
        console.log(startDateISOString);
        console.log("startDate.utc().startOf('day').toISOString()");
        console.log(startDate.utc().startOf('day').toISOString());

        let endDateISOString = `${endDate.format('YYYY-MM-DD')}T23:59:59.999Z`;
        console.log('#####################');
        console.log('endDate');
        console.log(endDate.toISOString());
        console.log('endDateISOString');
        console.log(endDateISOString);
        console.log("endDate.utc().endOf('day').toISOString()");
        console.log(endDate.utc().endOf('day').toISOString());
        
        return startDateISOString + '/' + endDateISOString;
    }
```

returns
```javascript
#####################
startDate
2018-04-30T22:00:00.000Z
startDateISOString
2018-05-01T00:00:00.000Z
startDate.utc().startOf('day').toISOString()
2018-04-30T00:00:00.000Z
#####################
endDate
2018-05-07T21:59:59.999Z
endDateISOString
2018-05-07T23:59:59.999Z
endDate.utc().endOf('day').toISOString()
2018-05-07T23:59:59.999Z
```

Reply:
All right, but did you try the first option (without utc)?

```js
const start = moment().startOf('day'); // set to 12:00 am today
const end = moment().endOf('day'); // set to 23:59 pm today
```

Reply:
They don't change the value of startDate and endDate because they are already non UTC endOfDay and startOfDay

Reply:
Alright, thanks for clarrifying